### PR TITLE
Update tests with `.test` prefix structure for simplified grepping

### PR DESCRIPTION
**Pull request by damassi**
This PR updates our test filenames to match a new `filename.test.coffee` pattern, enabling us to simplify how we find and execute said tests:

```sh
$(BIN)/mocha $(shell find api -name '*.test.coffee')
$(BIN)/mocha $(shell find client -name '*.test.coffee')
```
(No more `/*/*/*/test` and exclusion matching required.)

### CODE REVIEW

**Comment by kanaabe** [comment]
No worries! Thanks for the quick fix 👍

**Comment by kanaabe** [comment]
So down for the extensions -- wanna update the Makefile as well?

**Comment by damassi** [comment]
Yup yup, on it now 👍

File: Makefile [hunk 1]
```diff
@@ -14,17 +14,12 @@ s:
...
 test: assets
-	$(BIN)/mocha $(shell find api/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find api/apps/*/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find client/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find client/apps/*/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find client/apps/*/**/*/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find client/apps/*/**/*/**/test -name '*.coffee' -not -path 'test/helpers/*')
-	$(BIN)/mocha $(shell find client/apps/*/**/*/**/*/test -name '*.coffee' -not -path 'test/helpers/*')
+	$(BIN)/mocha $(shell find api -name '*.test.coffee')
+	$(BIN)/mocha $(shell find client -name '*.test.coffee')
```
Comment (damassi) [inline Makefile]: This change also clears the way to [co-locate your tests with your sources](https://github.com/artsy/emission/tree/master/src/lib/Components/Artist/__tests__) should you ever want to go that route.

**Pull request merged**